pam/pam.go

```diff
@@ -232,12 +233,30 @@ func (h *pamModule) handleAuthRequest(mode authd.SessionMode, mTx pam.ModuleTran
 
 	if mode == authd.SessionMode_PASSWD && flags&pam.PrelimCheck != 0 {
 		log.Debug(context.TODO(), "ChangeAuthTok, preliminary check")
-		_, closeConn, err := newClient(parsedArgs)
+		c, closeConn, err := newClient(parsedArgs)
 		if err != nil {
 			log.Debugf(context.TODO(), "%s", err)
 			return fmt.Errorf("%w: %w", pam.ErrTryAgain, err)
 		}
-		closeConn()
+		defer closeConn()
+
+		username, err := mTx.GetItem(pam.User)
+		if err != nil {
+			return pam.ErrAuthinfoUnavail
+		}
```

Comment:
```suggestion
		if username == "" || err != nil {
			return pam.ErrAuthinfoUnavail
		}
```

maybe? In general I'd return `err` as it is in case we receive one though as said in other comment.